File: Decorato.java/src/com/parttern/decorator/Jelly.java
package com.parttern.decorator;

public class Jelly extends ToppingDecorator {
    private double price = 0.1;

    public Jelly(MilkTea innerMilkTea) {
        super(innerMilkTea);
    }

    @Override
    public double cost() {
        return this.price + milkTea.cost();
    }

    @Override
    public String doMilkTea() {
        String type = milkTea.doMilkTea();
        return type + addJelly();
    }

    private String addJelly (){
        return " ,Thạch rau câu";
    }
}

File: Decorator/Jelly.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Decorator {
    internal class Jelly : ToppingDecorator {
        private readonly double price = 0.1;

        public Jelly(MilkTea milkTea) : base(milkTea) {
        }

        public override double Cost() {
            return price + milkTea.Cost();
        }

        public override string DoMilkTea() {
            return milkTea.DoMilkTea() + AddJelly();
        }

        private string AddJelly() {
            return ", Thạch rau câu";
        }
    }
}
